From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Blake Galbreath <blake.galbreath@gmail.com>
Date: Thu, 1 Jul 2021 19:25:05 -0500
Subject: [PATCH] Structure seed options


diff --git a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
index fde436065aeeb47cd177656a7c9fe8dc34178e87..da683b40c1963c529971573c80cadf0aa03a0639 100644
--- a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -417,6 +417,23 @@ public abstract class ChunkGenerator {
                     case "village":
                         seed = conf.villageSeed;
                         break;
+                    // Purpur start
+                    case "stronghold":
+                        if (net.pl3x.purpur.PurpurConfig.seedStructureStronghold != -1) {
+                            seed = net.pl3x.purpur.PurpurConfig.seedStructureStronghold;
+                        }
+                        break;
+                    case "mineshaft":
+                        if (net.pl3x.purpur.PurpurConfig.seedStructureMineshaft != -1) {
+                            seed = net.pl3x.purpur.PurpurConfig.seedStructureMineshaft;
+                        }
+                        break;
+                    case "buried_treasure":
+                        if (net.pl3x.purpur.PurpurConfig.seedStructureBuriedTreasure != -1) {
+                            seed = net.pl3x.purpur.PurpurConfig.seedStructureBuriedTreasure;
+                        }
+                        break;
+                    // Purpur end
                 }
 
                 updated.put(entry.getKey(), new StructureFeatureConfiguration(feature.spacing(), feature.separation(), seed));
diff --git a/src/main/java/net/pl3x/purpur/PurpurConfig.java b/src/main/java/net/pl3x/purpur/PurpurConfig.java
index ae34869f713b5cc2fabcb5521ce2e020e6a7c50b..99801676052591439056920e0a03f690869dee39 100644
--- a/src/main/java/net/pl3x/purpur/PurpurConfig.java
+++ b/src/main/java/net/pl3x/purpur/PurpurConfig.java
@@ -157,6 +157,17 @@ public class PurpurConfig {
         return builder.build();
     }
 
+    public static int seedStructureBuriedTreasure = -1;
+    public static int seedStructureMineshaft = -1;
+    public static int seedStructureStronghold = -1;
+    private static void seedSettings() {
+        seedStructureBuriedTreasure = getInt("settings.seed.structure.buried_treasure", seedStructureBuriedTreasure);
+        seedStructureMineshaft = getInt("settings.seed.structure.mineshaft", seedStructureMineshaft);
+        seedStructureStronghold = getInt("settings.seed.structure.stronghold", seedStructureStronghold);
+        // hide these from timings report
+        if (!TimingsManager.hiddenConfigs.contains("settings.seed")) TimingsManager.hiddenConfigs.add("settings.seed");
+    }
+
     public static String serverModName = "Purpur";
     private static void serverModName() {
         serverModName = getString("settings.server-mod-name", serverModName);
